src/main/java/seedu/address/logic/parser/SharkieParser.java

```diff
@@ -0,0 +1,112 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.commons.core.Messages.MESSAGE_UNKNOWN_COMMAND;
+import static seedu.address.logic.parser.CliPrefix.*;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import seedu.address.logic.commands.people.PeopleAddCommand;
+import seedu.address.logic.commands.people.PeopleClearCommand;
+import seedu.address.logic.commands.Command;
+import seedu.address.logic.commands.people.PeopleDeleteCommand;
+import seedu.address.logic.commands.people.PeopleEditCommand;
+import seedu.address.logic.commands.sharkie.ExitCommand;
+import seedu.address.logic.commands.people.PeopleFindCommand;
+import seedu.address.logic.commands.sharkie.HelpCommand;
+import seedu.address.logic.commands.people.PeopleListCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.logic.parser.people.PeopleAddCommandParser;
+import seedu.address.logic.parser.people.PeopleDeleteCommandParser;
+import seedu.address.logic.parser.people.PeopleEditCommandParser;
+import seedu.address.logic.parser.people.PeopleFindCommandParser;
+
+/**
+ * Parses user input.
+ */
+public class SharkieParser {
+
+    /**
+     * Used for initial separation of command word and args.
+     */
+    private static final Pattern BASIC_COMMAND_FORMAT =
+            Pattern.compile("(?<commandType>\\S+) ?(?<commandWord>\\S+)?(?<arguments>.*)");
+
+    /**
+     * Parses user input into command for execution.
+     *
+     * @param userInput full user input string
+     * @return the command based on the user input
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public Command parseCommand(String userInput) throws ParseException {
+        final Matcher matcher = BASIC_COMMAND_FORMAT.matcher(userInput.trim());
+        if (!matcher.matches()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, HelpCommand.MESSAGE_USAGE));
+        }
+
+        final String commandType = matcher.group("commandType");
+        final String commandWord = matcher.group("commandWord");
+        final String arguments = matcher.group("arguments");
+
+        switch(commandType) {
+        case PEOPLE_COMMAND_TYPE:
+            return getPeopleCommand(commandWord, arguments);
+        case WALLET_COMMAND_TYPE:
+            return getWalletCommand(commandWord, arguments);
+        case SHARKIE_COMMAND_TYPE:
+            return getSharkieCommand(commandWord, arguments);
+        default:
+            throw new ParseException(MESSAGE_UNKNOWN_COMMAND);
```

Comment:
Maybe this should be a different message to the user because it's not an unknown command but an unknown command type/prefix

Reply:
Probably can leave it as it is because we are doing "help" and "exit" so it makes sense if it isn't a proper command